From 3511ffec0fa6379c827c203cc7f31fc95f693e99 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:08:35 -0400
Subject: [powerpc] pseries: Select ARCH_RANDOM on pseries

Message-id: <1410545655-205645-287-git-send-email-dzickus@redhat.com>
Patchwork-id: 94220
O-Subject: [RHEL7 PATCH 286/626] powerpc/pseries: Select ARCH_RANDOM on pseries
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 8d4887ee30695d6b092e96029ae974505c43bfb6
Author: Michael Ellerman <mpe@ellerman.id.au>
Date:   Tue Dec 10 14:02:49 2013 +1100

    powerpc/pseries: Select ARCH_RANDOM on pseries

    We have a driver for the ARCH_RANDOM hook in rng.c, so we should select
    ARCH_RANDOM on pseries.

    Without this the build breaks if you turn ARCH_RANDOM off.

    This hasn't broken the build because pseries_defconfig doesn't specify a
    value for PPC_POWERNV, which is default y, and selects ARCH_RANDOM.

    Signed-off-by: Michael Ellerman <mpe@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/pseries/Kconfig b/arch/powerpc/platforms/pseries/Kconfig
index b8eac45..2cb8b77 100644
--- a/arch/powerpc/platforms/pseries/Kconfig
+++ b/arch/powerpc/platforms/pseries/Kconfig
@@ -20,6 +20,7 @@ config PPC_PSERIES
  select PPC_DOORBELL
  select HAVE_CONTEXT_TRACKING
  select HOTPLUG_CPU if SMP
+ select ARCH_RANDOM
  default y
 
 config PPC_SPLPAR
-- 
1.7.1